sdk/messaging/azservicebus/message.go

```diff
@@ -14,53 +15,81 @@ import (
 	"github.com/devigned/tab"
 )
 
-type (
-	// ReceivedMessage is a received message from a Client.NewReceiver().
-	ReceivedMessage struct {
-		Message
+// ReceivedMessage is a received message from a Client.NewReceiver().
+type ReceivedMessage struct {
+	MessageID string
 
-		LockToken              [16]byte
-		DeliveryCount          uint32
-		LockedUntil            *time.Time // `mapstructure:"x-opt-locked-until"`
-		SequenceNumber         *int64     // :"x-opt-sequence-number"`
-		EnqueuedSequenceNumber *int64     // :"x-opt-enqueue-sequence-number"`
-		EnqueuedTime           *time.Time // :"x-opt-enqueued-time"`
-		DeadLetterSource       *string    // :"x-opt-deadletter-source"`
+	ContentType      string
```

Comment:
Is there a functional reason not to embed `Message`?

Reply:
Yes, and it's all just to make it so you have to use `ReceivedMessage.Body() ([]byte, error)`:

(longer explanation):

It is possible for a user to send an AMQP message directly (ie, bypassing any of the convenience encoding). We don't have this exported in our SDK but all the other track 2 SDKs for Service Bus do (via AMQPAnnotatedMessage).

If they do this the 'Body' field can be misleading - it either won't contain a value (because the payload has been encoded differently than expected) or we could try to put something in there but then the user wouldn't have any way of knowing that something is wrong and it'd be an approximation.